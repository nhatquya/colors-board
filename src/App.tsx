import React from "react";
import { DndProvider } from "react-dnd";
import Backend from "react-dnd-html5-backend";

import Board from "./Board";

const wrapperStyle: React.CSSProperties = {
  textAlign: "center",
};

const containerStyle = {
  width: 500,
  height: 500,
  border: "1px solid gray",
  margin: "0 auto",
};

function App() {
  return (
    <div style={wrapperStyle}>
      <h2>Colors Board</h2>
      <div style={containerStyle}>
        <DndProvider backend={Backend}>
          <Board />
        </DndProvider>
      </div>
    </div>
  );
}

export default App;
