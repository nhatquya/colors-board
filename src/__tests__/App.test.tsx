import React from "react";
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json';
import App from "../App";

jest.mock('../Board', () => (props: any) => <mock-board {...props} />);

it("should render App correctly", () => {
  const tree = mount(<App />);
  expect(toJSON(tree)).toMatchSnapshot();
});
