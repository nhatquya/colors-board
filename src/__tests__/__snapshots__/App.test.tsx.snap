// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<App>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h2>
      Colors Board
    </h2>
    <div
      style={
        Object {
          "border": "1px solid gray",
          "height": 500,
          "margin": "0 auto",
          "width": 500,
        }
      }
    >
      <DndProvider
        backend={[Function]}
      >
        <Component>
          <mock-board />
        </Component>
      </DndProvider>
    </div>
  </div>
</App>
`;
